controllers/books.py

```diff
@@ -245,7 +278,7 @@ def dummy():
         questions = _exercises(base_course, chapter)
     logger.debug("QUESTIONS = {} {}".format(subchapter, questions))
     return dict(
-        course_name=course.course_name,
+        course_name=current_course,
```

Comment:
I don't think this is assigned for all cases.

Reply:
I don't know what you mean?

I am interested in what you think of the concept of this PR.

Reply:
I misunderstood -- was giving more of a code review. I like the concept.

I meant, does `current_course` get assigned for all cases? Glancing through the code, I only see its being assigned inside an `if` statement, but I might have missed something.

Reply:
I think it is good.

if course is not found then we set it.
but if course is not found then we are either going to set it or redirect the user back to the change course page.

Reply:
Except for the case where the user is not logged in! 
Caught by the unit tests too!

Reply:
+1 for unit tests!